source/common/quic/envoy_quic_utils.cc

```diff
@@ -285,9 +285,8 @@ void adjustNewConnectionIdForRouting(quic::QuicConnectionId& new_connection_id,
                                      const quic::QuicConnectionId& old_connection_id) {
   char* new_connection_id_data = new_connection_id.mutable_data();
   const char* old_connection_id_ptr = old_connection_id.data();
-  auto* first_four_bytes = reinterpret_cast<const uint32_t*>(old_connection_id_ptr);
   // Override the first 4 bytes of the new CID to the original CID's first 4 bytes.
-  safeMemcpyUnsafeDst(new_connection_id_data, first_four_bytes);
+  memcpy(new_connection_id_data, old_connection_id_ptr, 4); // NOLINT(safe-memcpy)
```

Comment:
Would replacing the reinterpret_cast with `safeMemcpyUnsafeSrc(&first_four_bytes, old_connection_id_ptr)` work?

Reply:
That would only get you one byte I think.

Reply:
Yeah, exactly. safeMemcpyUnsafeDst infers the length of the data to copy by the size of the type passed in.

Reply:
Thanks, Dan! I definitely misread your first suggestion.